edx_exams/apps/api/v1/tests/test_views.py

```diff
@@ -513,3 +513,182 @@ def test_proctoring_providers_list_empty(self):
+        """
+        headers = self.build_jwt_headers(user)
+        return self.client.get(self.url, **headers)
+
+    def test_auth_failures(self):
+        """
+        Verify the endpoint validates permissions
+        """
+        # Test unauthenticated
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 401)
+
+    def test_access_not_granted(self):
+        """
+        Verify the endpoint doesn't grant access for an exam
+        without an existing exam attempt or past due date.
+        """
+        response = self.get_exam_access(self.user)
+        self.assertEqual(403, response.status_code)
+
+    def test_access_granted_exam_attempt(self):
+        """
+        Verify the endpoint grants access for an exam
+        with an existing exam attempt.
+        """
+        start_time = self.due_date - timedelta(minutes=60)
+        allowed_time_limit_mins = self.exam.time_limit_mins
+        ExamAttempt.objects.create(
+            user=self.user,
+            exam=self.exam,
+            attempt_number=1,
+            status='started',
+            start_time=start_time,
+            allowed_time_limit_mins=allowed_time_limit_mins
+        )
+
+        response = self.get_exam_access(self.user)
+        self.assertEqual(200, response.status_code)
+        self.assertIsInstance(response.cookies["exam_access_ticket"].value, str)
+
+    def test_access_granted_not_started_exam_attempt(self):
+        """
+        Verify the endpoint grants access for an exam
+        with an existing exam attempt that is not started.
+        """
+        start_time = self.due_date - timedelta(minutes=60)
+        allowed_time_limit_mins = self.exam.time_limit_mins
+        ExamAttempt.objects.create(
+            user=self.user,
+            exam=self.exam,
+            attempt_number=1,
+            start_time=start_time,
+            allowed_time_limit_mins=allowed_time_limit_mins
+        )
+
+        response = self.get_exam_access(self.user)
+        self.assertEqual(200, response.status_code)
+        self.assertIsInstance(response.cookies["exam_access_ticket"].value, str)
+
+    def test_access_granted_exam_attempt_zero_time_remaining(self):
+        """
+        Verify the endpoint grants access for an exam
+        with an existing exam attempt with zero time remaining.
+        """
+        start_time = self.due_date
+        allowed_time_limit_mins = self.exam.time_limit_mins
+        ExamAttempt.objects.create(
+            user=self.user,
+            exam=self.exam,
+            attempt_number=1,
+            status='started',
+            start_time=start_time,
+            allowed_time_limit_mins=allowed_time_limit_mins
+        )
+
+        response = self.get_exam_access(self.user)
+        self.assertEqual(200, response.status_code)
+        self.assertIsInstance(response.cookies["exam_access_ticket"].value, str)
+        # todo: also check that exp is correct in cookie?
```

Comment:
Question: Should I also check that the exp window is correct? If so, is it better to only test the helper method that calculates this (set_expiration_window) or would it be better to actually test that the expiration is correct in the cookie by decoding and checking?